test/e2e/controller/main_test.go

```diff
@@ -15,12 +15,12 @@
 package controller
 
 import (
-	"log"
 	"os"
 	"testing"
 
 	e2eframework "agones.dev/agones/test/e2e/framework"
 	"github.com/sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
```

Comment:
So this is a several cross cutting pieces in one PR.
- Change logging to to Logrus
- Implement pflag over flag (I think I mentioned we should tackle this at a later date?)
- Pass Feature_Gates

For future references, these should be separate PRs, having this all together makes things harder to review and track the changes.

Reply:
I agree. That was a misunderstanding from my side, I considered your note as "nice to add" in a current scope of changes.